core/src/test/scala/unit/kafka/server/AbstractApiVersionsRequestTest.scala

```diff
@@ -34,18 +36,25 @@ import scala.jdk.CollectionConverters._
 abstract class AbstractApiVersionsRequestTest(cluster: ClusterInstance) {
 
   def sendApiVersionsRequest(request: ApiVersionsRequest, listenerName: ListenerName): ApiVersionsResponse = {
-    IntegrationTestUtils.connectAndReceive[ApiVersionsResponse](request, cluster.brokerSocketServers().asScala.head, listenerName)
+    val socket = if (listenerName == controlPlaneListenerName) {
+      cluster.controllerSocketServers().asScala.head
+    } else {
+      cluster.brokerSocketServers().asScala.head
+    }
+    IntegrationTestUtils.connectAndReceive[ApiVersionsResponse](request, socket, listenerName)
   }
 
   def controlPlaneListenerName = new ListenerName("CONTROLLER")
```

Comment:
Not from this patch, but I found the overloading here a bit confusing. The control plane listener is only relevant to zk clusters, but here we're using the CONTROLLER name so that we can also use it to designate the controller for kraft clusters (as in `sendApiVersionsRequest` above). I wonder if it would be clearer to have two separate methods:
```scala
def controlPlaneListener: Option[ListenerName] = if (cluster.isKRaftTest) None else Some(new ListenerName("CONTROL_PLANE"))

def controllerListener: Option[ListenerName] = if (cluster.isKRaftTest) Some(new ListenerName("CONTROLLER") else None
```
Then maybe we could have separate test cases:

```scala
   @ClusterTest(clusterType = ClusterType.ZK)
   def testApiVersionsRequestThroughControlPlaneListener(): Unit = ???

   @ClusterTest(clusterType = ClusterType.RAFT)
   def testApiVersionsRequestThroughControllerListener(): Unit = ???
```

What do you think?